### PR TITLE
HTK - Sharing Updates

### PR DESCRIPTION
**Pull request by StephenHodgson**
Fixes and updates for some of the sharing features.

- Added Troubleshooting section to readme. [Preview Here](https://github.com/HodgsonSDAS/HoloToolkit-Unity/tree/HTK-Sharing/Assets/HoloToolkit/Sharing#troubleshooting)
- Fixed bug where every user was named _User0_.  Users are now named after the device, the environment user name, or the user id assigned by the sharing service.
- Fixed build warning about uninitialized variable `spawnablePrefabs`.
- Added a text mesh to the `SharingTest.scene` that shows anchor debug info.
- Added an ability to turn on and off detailed logs for sharing service specific components.
- Added `IsConnected` bool in `SharingStage` that checks for a valid `Manager` and `Connection` before returning `Connection.IsConnected()`.
- Fixed issue where components that listen for Session connections are never activated after connection event has been raised.  Components will now properly check if there is already a session connection, then register for the delegate if no connection exists.
- `ImportExportAnchorManager` now works in editor for easier testing, and faster iteration.  `UNITY_WSA && !UNITY_EDITOR` wraps the appropriate methods, fields, and parameters.  No public or or external facing code was changed/deprecated.
- `SharingSpawnTest.scene` now works in Editor and on device. Partially resolves https://github.com/Microsoft/HoloToolkit-Unity/issues/494. 
- `SharingSpawnTest.scene` now shows how to both create and delete sync spawn objects. Resolves https://github.com/Microsoft/HoloToolkit-Unity/issues/517.

### CODE REVIEW

File: Assets/HoloToolkit/Sharing/README.md [hunk 1]
```diff
@@ -96,14 +97,28 @@ Any SyncPrimitive tagged with the [SyncData] attribute will automatically be add
...
 1. SpatialPerception
-2. InternetClientServer
-3. PrivateNetworkClientServer
-4. Microphone capabilities
+2. InternetClient
+3. InternetClientServer
+4. PrivateNetworkClientServer
+5. Microphone capabilities
+
+Enabling the Sharing Service will also uppack a new directory in your projects root folder named `External`.
```
Comment (NeerajW) [inline Assets/HoloToolkit/Sharing/README.md]: Typo: unpack

File: Assets/HoloToolkit/Sharing/Scripts/SharingStage.cs [hunk 2]
```diff
@@ -128,7 +135,7 @@ public string UserName
             }
             set
             {
-                using (XString userName = new XString(value))
+                using (var userName = new XString(value))
```
Comment (NeerajW) [inline Assets/HoloToolkit/Sharing/Scripts/SharingStage.cs]: I am surprised you changed it to var ;) :)

File: Assets/HoloToolkit/Sharing/Scripts/SharingStage.cs [hunk 3]
```diff
@@ -245,7 +268,7 @@ private void LateUpdate()
 
         private void Connect()
         {
-            ClientConfig config = new ClientConfig(ClientRole);
+            var config = new ClientConfig(ClientRole);
```
Comment (NeerajW) [inline Assets/HoloToolkit/Sharing/Scripts/SharingStage.cs]: Why are these being changed?

File: Assets/HoloToolkit/Sharing/Scripts/SharingStage.cs (hunk 2 above)
Comment (HodgsonSDAS) [inline Assets/HoloToolkit/Sharing/Scripts/SharingStage.cs]: Haha, I updated my formatting wizard to catch when the type is evident, like when creating new instances.  You're pretty much writing it again after declaring the type. I think it's fine as long as it's readable.

File: Assets/HoloToolkit/Sharing/Scripts/SharingStage.cs (hunk 3 above)
Comment (HodgsonSDAS) [inline Assets/HoloToolkit/Sharing/Scripts/SharingStage.cs]: Same reason as above.  The formatting wizard caught it.

**Comment by HodgsonSDAS** [comment]
Can I get some love on this PR?

**Comment by HodgsonSDAS** [comment]
Can I get a second reviewer to test this out please? I'd like to get this in soon.

**Pull request merged**